metadata-ws/src/main/java/uk/ac/ebi/ampt2d/metadata/aop/ReleaseDateAspect.java

```diff
@@ -72,13 +81,38 @@ public Object filterOnReleaseDateAdviceFindOne(ProceedingJoinPoint proceedingJoi
     }
 
     private Object getObjectUnlessBeforeReleaseDate(Object result) {
-        if (result instanceof Auditable) {
-            LocalDate releaseDate = ((Auditable) result).getReleaseDate();
-            if (releaseDate != null && releaseDate.isAfter(LocalDate.now())) {
-                return null;
-            }
+        if (!(result instanceof Auditable)) {
+            return result;
+        }
+
+        LocalDate releaseDate = ((Auditable) result).getReleaseDate();
+        if (releaseDate == null || !releaseDate.isAfter(LocalDate.now())) {
+            return result;
+        }
+
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+
+        if (authentication instanceof AnonymousAuthenticationToken) {
+            throw new SecurityException("Authentication required");
+        }
+
+        //Allow admins to access before release date
+        if (authentication.getAuthorities().stream()
+                .anyMatch(authority -> authority.getAuthority()
+                        .equals("ROLE_" + EnableSecurityConfig.ROLE_SERVICE_OPERATOR))) {
+            return result;
+        }
+
+        String commaSeperatedPermittedStudiesToTheUsers = ((CustomUsernamePasswordAuthenticationToken)
+                ((OAuth2Authentication) authentication).getUserAuthentication()).getStudies();
+        List<String> permittedStudies = (commaSeperatedPermittedStudiesToTheUsers.isEmpty()) ? new ArrayList<>() : Arrays.asList
+                (commaSeperatedPermittedStudiesToTheUsers.split(","));
+        String studiesAssociatedToEntities = ((Auditable) result).getStudyIds();
+        if (permittedStudies.stream()
+                .anyMatch(permittedStudy -> studiesAssociatedToEntities.contains(permittedStudy))) {
```

Comment:
OK, so this way of doing it with a comma-separated list of studies is acceptable; but is there definitely no way to specify studies as separate resources in Keycloak? It would be more user friendly to whomever is going to be ultimaterly responsible for managing the user permissions

Reply:
the feature which we are using in Keycloak for this functionality is kind of key-value pairs, so key being the userId values is kinda made a comma-separated list of study accessions.

Reply:
OK, understood. Anyway, the current implementation is very reasonable so no point of trying to optimise is further (especially considering EGA use case is far from deployment at this point)

Reply:
Sorry, I explained it wrong but it is the same case just the key is a studies which will be included in the jwt token.